### DISCUSSION TITLE
How to obtain the nits from the RGB value?

### DISCUSSION THREAD

**Discussion by chesschi** [colour-science/colour#717] [Q&A] [answered]
A while ago I saw @Nick-Shaw has replied in this [forum](https://www.liftgammagain.com/forum/index.php?threads/convert-rgb-values-to-hdr-nit-values.9184/) that colour-science git repo provides several nice features such as `eotf_ST2084` and `oetf_ST2084` to convert values between nits and XYZ values.

In this forum, Pepijn Klijs mentioned that the LightSpace tool provides a nice GUI to convert the RGB values to the nits value. Are there any equivalent function provided by colour-science to do this RGB to nits conversion?

Many thanks!

**Comment by nick-shaw** [comment, marked as answer]
Taking an example 10 bit RGB triple, [ 123, 456, 789 ] you would use the following code:
```>>> from __future__ import division
>>> import colour
>>> import numpy as np
>>> RGB_to_XYZ_matrix = colour.models.BT2020_COLOURSPACE.RGB_to_XYZ_matrix
>>> RGB_CV = np.array( [ 123, 456, 789 ] )
>>> RGB_lin = colour.models.eotf_ST2084(RGB_CV / 1023)
>>> colour.utilities.dot_vector(RGB_to_XYZ_matrix, RGB_lin)
array([  209.81942735,   106.90999103,  1269.42240872])
```
The luminance is the Y of XYZ, so the second component above, which is 106.9 nits in this example.

**Comment by chesschi** [comment]
What a nice library, brilliant! Many thanks!

**Comment by nick-shaw** [comment]
Add `colour.XYZ_to_xyY` onto the end of that sequence, and you get the chromaticity coordinates as well as the luminance.
